### ISSUE TITLE
Support for GRPC interractions

### ISSUE THREAD

**Issue by dedoussis** [kevin1024/vcrpy#560]
This issue is about adding a stub that wraps the [https://pypi.org/project/grpcio/](url) package to support cassettes for GRPC interactions.

Submitting this issue to see whether a contribution for supporting such a feature would be welcome. If yes, I'm happy to raise a PR.

**Comment by jaoxford** [comment]
Also having the same issue.

We are trying to test Google Cloud Vision interactions, which uses gPRC under the hood.

**Comment by maxcountryman** [comment]
I'm also interested in this for testing Google Ads interactions, which also utilize gRPC.

Is this something that would be pulled into vcrpy?

**Comment by ServerlessArchitect** [comment]
Count me in for alpha testing :)

**Comment by abdullahselek** [comment]
Interested in gRPC support too 👍

**Comment by upstartjohnvandivier** [comment]
@dedoussis there seems to be a good amount of support, are you able to go ahead and make a PR to nudge the convo along?

cc @kevin1024 in case I missed convo on this elsewhere or we are waiting on a maintainer decision before pr

**Comment by agilebean** [comment]
any update on this?
I really need grpc support for accessing Google texttospeech and speechtotext API.

**Comment by deividasskiparis** [comment]
Any news on this? 🙏 🤞

**Comment by sbouchardet** [comment]
Any news here? 🤞 
did someone find a workaround?

**Comment by jkline-trustle** [comment]
I also request VCR please support gRPC.
My use case is recording interactions with Google Cloud PubSub using [google-cloud-pubsub library](https://cloud.google.com/python/docs/reference/pubsub/latest).

**Comment by kevin1024** [comment]
Closing as wontfix. gRPC uses HTTP/2 with protobuf framing, which is architecturally outside vcrpy's interception model (it hooks stdlib `http.client`, urllib3, httpx, aiohttp, etc.). Supporting it would require a fundamentally different recording mechanism that isn't on the roadmap.

For Google Cloud client libraries, the common workaround is to force the REST transport (e.g. `transport="rest"`) so requests go over HTTP and can be recorded by vcrpy. If someone wants to champion a gRPC interceptor-based approach as a separate integration, a PR or proposal is welcome and we can revisit. Thanks for the interest and patience on this long-standing request!
